### DISCUSSION TITLE
Append utm=14 to all google searches

### DISCUSSION THREAD

**Discussion by cvk14** [mshibanami/redirect-web#31] [Questions] [answered]
Given recent news about the ability to append utm=14 to google searches to disable the AI portions the google search, How can i set a rule to accept any google search and append &utm=14 at the end of the search?

**Comment by mshibanami** [comment, marked as answer]
@cvk14 
It's `udm=14`, and yes! You can create a rule that auto-appends it, like this:

- **Redirect From**:
    - **URL Pattern**:
    ``` regex
    https?:\/\/(?:www\.)?google\.(?:com|com\.[a-z]{2}|co\.[a-z]{2}|[a-z]{2})\/search\?((.*&)?q=.*)$
    ```
    - **Pattern Type**: Regular Expression
- **Redirect To**: `$0&udm=14`
- **Excluded URL Pattern**
    - URL Pattern: 
    ```regex
    .*[?&](tbm=|udm=).*
    ```
    - **Pattern Type**: Regular Expression

Regarding the Excluded URL Pattern > URL Pattern:

- `tbm=`: This is for all the tabs except All and Images, such as Videos, News, etc.
- `udm=`: This seems to be another parameter for specifying the tab, and I specify it for these reasons:
    - For the Images tab, which include `udm=2` instead of `tbm`.
    - Other tabs that Google will add in the future.
    - To avoid an infinite loop of appending `udm=14`

----

To be honest, I've never seen this AI feature on Google Search. But I'll already add this rule to the library, which is coming soon!

**Reply by mshibanami** [reply]
> To be honest, I've never seen this AI feature on Google Search.

I think I understand it now. The parameter `udm=14` is used in the Web tab, which is hidden by default. The difference between the All tab and the Web tab isn’t necessarily related to AI but rather to whether it includes more rich/redundant features. For example, in this case, the All tab shows a quote from a website and the “People also ask” section while the Web tab doesn’t show these:

| Without `udm=14` | With `udm=14` |
|-|-|
|![all](https://github.com/mshibanami/redirect-web/assets/1333214/4ca10659-a93c-4c86-a994-05556c7a13c7)|![web](https://github.com/mshibanami/redirect-web/assets/1333214/bc477f3f-4e73-429c-901d-c1d3cb60f57e)|
